onnxruntime/core/framework/utils.cc

```diff
@@ -637,10 +637,12 @@ common::Status ExecuteGraph(const SessionState& session_state,
 #ifdef ENABLE_TRAINING
 common::Status ExecutePartialGraph(const SessionState& session_state, FeedsFetchesManager& feeds_fetches_manager,
                                    const std::vector<OrtValue>& feeds, std::vector<OrtValue>& fetches,
-                                   const logging::Logger& logger, PartialGraphExecutionState& state) {
+                                   const logging::Logger& logger, PartialGraphExecutionState& state,
+                                   OrtValueCache* cache) {
```

Comment:
>OrtValueCache* cache [](http://example.com/codeflow?start=35&length=20)

Please do not use raw pointers, instead use shared or unique ptr or pass by reference.